sklearn/metrics/classification.py

```diff
@@ -186,6 +187,74 @@ def accuracy_score(y_true, y_pred, normalize=True, sample_weight=None):
     return _weighted_sum(score, sample_weight, normalize)
 
 
+def multilabel_confusion_matrix(y_true, y_pred):
+    """Compute True positive, False positive, True negative, False negative
+    for a multilabel classification problem.
+
+    Parameters
+    ----------
+    y_true : array-like or scipy sparse matrix, shape = (n_samples, n_labels)
+        Ground truth (correct) target label(s).
+
+    y_pred : array-like or scipy sparse matrix, shape = (n_samples, n_labels)
+        Estimated target label(s) as returned by a classifier.
+
+    Returns
+    -------
+    MCM : ndarray of shape = (n_labels, 4)
+        The multilabel confusion matrix.
+
+        Every row contains the 4 confusion values for that particular label.
+        The confusion values are in the following order - tp, fp, tn and fn.
+
+        ``MCM[i, :]`` returns the various confusion values for label i and
+        ``MCM[:, 0]`` returns the true positive values for all the labels.
+
+    References
+    ----------
+    .. [1] `Wikipedia entry for the Confusion matrix
+           <http://en.wikipedia.org/wiki/Sensitivity_and_specificity>`_
+    .. [2] http://www.cnts.ua.ac.be/~vincent/pdf/microaverage.pdf
+
+    Examples
+    --------
+    >>> from sklearn.metrics import multilabel_confusion_matrix
+    >>> y_true = np.array([[1, 0, 0], [1, 0, 0]])
+    >>> y_pred = np.array([[0, 1, 1], [1, 0, 0]])
+    >>> multilabel_confusion_matrix(y_true, y_pred)[:, 0]
+    array([1, 0, 0])
+    """
+    if type(y_true) is list:
+        y_true = np.array(y_true)
+
+    if type(y_pred) is list:
+        y_pred = np.array(y_pred)
+
+    y_type, y_true, y_pred = _check_targets(y_true, y_pred)
+    if y_type != 'multilabel-indicator':
+        raise ValueError("%s is not supported" % y_type)
+
+    n_labels = y_true.get_shape()[1]
+
+    MCM = np.empty(shape=(n_labels, 4), dtype=int)
+
+    for label_idx in range(n_labels):
+        y_pred_col = y_pred.getcol(label_idx)
+        y_true_col = y_true.getcol(label_idx)
+        # tp can be obtained by dot product
+        t_pos = (y_true_col.transpose().dot(y_pred_col))[0, 0]
+        # fp are the ones in y_pred that
+        # match zeros in y_true
+        f_pos = count_nonzero(y_pred_col) - t_pos
+        f_neg = count_nonzero(y_true_col) - t_pos
+        zeros = y_true_col.get_shape()[0] - count_nonzero(y_true_col)
+        t_neg = zeros - f_pos
+
+        MCM[label_idx][:] = (t_pos, f_pos, f_neg, t_neg)
```

Comment:
Would it be better if this is ordered as a row major ordering of the regular confusion matrix? which will be tp, fn, fp, tn ? ( There seems to be no clear order followed in the literature )